File: lab3/src/main/java/lab3/myquest/items/BreakableItem.java
package lab3.myquest.items;

import lab3.functionality.Action;
import lab3.functionality.Item;

public class BreakableItem extends Item {

    public BreakableItem(String name, String description) {
        super(name, description);
        addAction(new Action("Разбить", p -> p.getName() + " задевает что-то своим телом. Бедная " + name, (q, p) -> {
            setName("Осколки");
            setDescription("Разбитые остатки какого-то предмета");
        }, true));
    }
}

File: lab4/src/main/kotlin/lab4/myQuest/items/BreakableItem.kt
package lab4.myQuest.items

import lab4.functionality.ActionFactory
import lab4.functionality.Description
import lab4.functionality.Item

class BreakableItem(name: String, description: Description) : Item(name, description) {
    init {
        actions.add(
            ActionFactory().changeNameDescription("Разбить",
                { "${it.name} задевает что-то своим телом. Бедная $name" },
                "Осколки",
                { "Разбитые остатки какого-то предмета" })
        )
    }
}
